### PR TITLE
feat(textfield): helperTextContent setter

### LINKED ISSUE #1470: Standard to display warnings
## Feature Requests

We need a standard to define how to display warnings on input fields, such as CAPS lock warnings on password fields.

### PR DESCRIPTION
**Pull request by patrickrodee**
Adds an MDCTextField accessor `helptextValue` to get/set the `textContent` of the help text field. Tests have been added as well as readme documentation. I tried to make the accessor name `helptextValue` similar to the existing `helptextElement` but the name `helptextValue` is a little verbose. I'd love some feedback and am totally open to changing the accessor name.

Fixes #1470

### CODE REVIEW

File: packages/mdc-textfield/index.js [hunk 1]
```diff
@@ -113,6 +113,14 @@ class MDCTextfield extends MDCComponent {
     this.foundation_.setValid(valid);
   }
 
+  get helptextValue() {
```
Comment (bonniezhou) [inline packages/mdc-textfield/index.js]: Remember to annotate for Closure:  https://material.io/components/web/docs/closure-compiler/

File: packages/mdc-textfield/index.js [hunk 2]
```diff
@@ -249,6 +257,17 @@ class MDCTextfield extends MDCComponent {
...
+      setHelptextValue: (value) => {
+        if (this.helptextElement) {
+          this.helptextElement.textContent = value;
+        }
+      },
+      getHelptextValue: () => {
+        if (!this.helptextElement) {
+          return;
+        }
+        return this.helptextElement.textContent.trim();
```
Comment (bonniezhou) [inline packages/mdc-textfield/index.js]: What's the reason for adding `trim()`? I assume users would want to know if they have extra whitespace.

File: packages/mdc-textfield/README.md [hunk 3]
```diff
@@ -373,6 +373,11 @@ being used as the text field's "help text". It defaults to `null`. If the text f
 contains an `aria-controls` attribute on instantiation of the component, it will look for an element
 with the corresponding id within the document and automatically assign it to this property.
 
+##### MDCTextfield.helptextValue
+
+String. Proxies to the foundation's `getHelptextValue/setHelptextValue` methods when retrieved/set
```
Comment (amsheehan) [inline packages/mdc-textfield/README.md]: nit: should read `getHelptextValue`/`setHelptextValue`

I realize you were just following the pattern for `MDCTextfield.disabled`, but that should probably change too.

File: test/unit/mdc-textfield/foundation.test.js [hunk 4]
```diff
@@ -173,6 +174,28 @@ test('#init does not add mdc-textfield__label--float-above class if the input do
   td.verify(mockAdapter.addClassToLabel(cssClasses.LABEL_FLOAT_ABOVE), {times: 0});
 });
 
+test('#setHelptextValue has no effect when no help text element is present', () => {
+  const {foundation} = setupTest();
+  foundation.setHelptextValue('foo');
```
Comment (amsheehan) [inline test/unit/mdc-textfield/foundation.test.js]: Hmmmm... I wonder if there's a way to make this a more isolated and specific test. It's not _really_ testing what we're saying it's testing.

File: test/unit/mdc-textfield/foundation.test.js [hunk 5]
```diff
@@ -173,6 +174,28 @@ test('#init does not add mdc-textfield__label--float-above class if the input do
...
+
+test('#setHelptextValue sets the value of the help text element', () => {
+  const {foundation, mockAdapter} = setupTest();
+  foundation.setHelptextValue('foo');
+  td.verify(mockAdapter.setHelptextValue('foo'));
+});
+
+test('#getHelptextValue has no effect when no help text element is present', () => {
+  const {foundation} = setupTest();
+  foundation.getHelptextValue();
```
Comment (amsheehan) [inline test/unit/mdc-textfield/foundation.test.js]: Same comment as above

File: test/unit/mdc-textfield/mdc-textfield.test.js [hunk 6]
```diff
@@ -131,6 +131,22 @@ test('set valid updates the component styles', () => {
...
+  component.helptextElement = helptext;
+  component.helptextValue = 'foo';
+  assert.equal(helptext.textContent, 'foo');
+});
+
+test('get helptextValue retrieves the help text element content', () => {
+  const {component} = setupTest();
+  const helptext = getHelptext();
+  component.helptextElement = helptext;
+  helptext.textContent = 'foo';
```
Comment (amsheehan) [inline test/unit/mdc-textfield/mdc-textfield.test.js]: I don't think we need to set anything here first, right? The fixture has a value already.

**Comment by amsheehan** [comment]
And some conflicts to resolve looks like.

File: packages/mdc-textfield/index.js (hunk 2 above)
Comment (patrickrodee) [reply packages/mdc-textfield/index.js]: Good point, I was trying to ensure that the returned value was close to what the user expected but that sounds like a false assumption on my part.

For example:
```
<p class="mdc-text-field-helptext">
  Hello
</p>
```
would have a `textContent` of `"  Hello "` when I would expect it to be just `"Hello"`

That being said, I see now that's an incorrect assumption. I'll remove `.trim()`. Thanks!

File: packages/mdc-textfield/index.js (hunk 1 above)
Comment (patrickrodee) [reply packages/mdc-textfield/index.js]: Good catch, thanks!

File: packages/mdc-textfield/README.md (hunk 3 above)
Comment (patrickrodee) [reply packages/mdc-textfield/README.md]: Easy fix, thanks!

File: test/unit/mdc-textfield/mdc-textfield.test.js (hunk 6 above)
Comment (patrickrodee) [reply test/unit/mdc-textfield/mdc-textfield.test.js]: Ah, good find. I'll switch the check for the fixture's default value. Thanks!

File: test/unit/mdc-textfield/foundation.test.js (hunk 4 above)
Comment (patrickrodee) [reply test/unit/mdc-textfield/foundation.test.js]: Good point. I moved that test to the `mdc-text-field.test.js` file.

https://github.com/patrickrodee/material-components-web/blob/37c078df5b21d597076636337e62a1a67494d59c/test/unit/mdc-textfield/mdc-text-field.test.js#L149-L153

File: test/unit/mdc-textfield/foundation.test.js (hunk 5 above)
Comment (patrickrodee) [reply test/unit/mdc-textfield/foundation.test.js]: Same as above, moved it to the `mdc-text-field.test.js` file

https://github.com/patrickrodee/material-components-web/blob/37c078df5b21d597076636337e62a1a67494d59c/test/unit/mdc-textfield/mdc-text-field.test.js#L162-L165

**Comment by patrickrodee** [comment]
Okie dokie, @bonniezhou and @amsheehan's review changes have been made!

File: packages/mdc-textfield/README.md [hunk 7]
```diff
@@ -373,7 +373,12 @@ being used as the text field's "helper text". It defaults to `null`. If the text
 contains an `aria-controls` attribute on instantiation of the component, it will look for an element
 with the corresponding id within the document and automatically assign it to this property.
 
-##### MDCTextField.disabled
+##### MDCTextfield.helperTextContent
```
Comment (bonniezhou) [inline packages/mdc-textfield/README.md]: Another renaming thing due to PR#1485 (sorry!)....MDCTextfield should be MDCTextField. Same for below.

Comment (patrickrodee) [reply packages/mdc-textfield/README.md]: Good eye! Fixed with af16d8d

**Comment by patrickrodee** [comment]
@bonniezhou getter has been removed

**Comment by patrickrodee** [comment]
CLA, please like me

**Comment by kfranqueiro** [comment]
Trying to ping @googlebot one more time, but commit authors LGTM

**Pull request merged**